packages/compiler/src/core/program.ts

```diff
@@ -910,3 +922,41 @@ function resolveOptions(options: CompilerOptions): CompilerOptions {
     outputPath: outputDir,
   };
 }
+
+async function emit(emitter: EmitterRef, options: CompilerOptions = {}, program: Program) {
+  const emitterName = emitter.metadata.name ?? "";
+  const outputRelativePath = options.listFiles
+    ? `./${getRelativePathFromDirectory(program.projectRoot, emitter.emitterOutputDir, false)}/`
+    : "";
+  addChildLog(`✓ ${emitterName}\t${outputRelativePath}`);
+  setChildLogPrefix(outputRelativePath);
+
+  const logger = createLogger({ sink: program.host.logSink });
+  if (logger.trackAction) {
```

Comment:
same here I think from the consumer point of view the trackAction should always be set on teh logger just hte logSink which is something hte user create themself potentially